compiler/noirc_evaluator/src/ssa/ir/instruction/binary.rs

```diff
@@ -116,13 +122,36 @@ pub(crate) fn eval_constant_binary_op(
             // Thus, we do not evaluate the division in this method, as we want to avoid triggering a panic,
             // and the operation should be handled by ACIR generation.
             if matches!(operator, BinaryOp::Div | BinaryOp::Mod) && rhs == 0 {
-                return None;
+                return Some(Err("attempt to divide by zero".to_string()));
             }
-            let result = function(lhs, rhs)?;
+
+            let Some(result) = function(lhs, rhs) else {
+                if let BinaryOp::Shl = operator {
+                    return None;
```

Comment:
Why is SHL not an overflow?

Reply:
Because of this: https://github.com/noir-lang/noir/issues/9022

I'll add a comment linking to that issue.